### PR TITLE
changes for task 8 regarding uploading a class image to the server

### PR DESCRIPTION
**Pull request by Engy-Mohamed**
task 8 :thank you

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 /public/build
 /public/hot
 /public/storage
+/public/assets/images/*
```
Comment (Nora-Lootfy) [inline .gitignore:6]: Ooooooh, you will hear me in the session telling them to add this to .gitignore just like Engy😂

File: app/Http/Controllers/ClassController.php [hunk 2]
```diff
@@ -32,11 +34,16 @@ public function store(Request $request)
...
             'class_name' => 'required|alpha_num:ascii|max:255',
-            'capacity' => 'required|numeric|integer|min:1', 
+            'capacity' => 'required|numeric|integer|min:1',
             'price' => 'required|decimal:0,2',
             'time_From' => 'required|date|after:tomorrow',
             'time_to' => 'required|date|after:time_From',
+            'image' => 'required|mimes:png,jpg,jpeg|max:2048',
         ]);
+
+        $file_name = $this->uploadFile($request['image'], 'assets/images');
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ClassController.php:44]: 100 out of 10 ya Engy, thanks for your hard working😊

File: app/Http/Controllers/ClassController.php [hunk 3]
```diff
@@ -67,11 +75,16 @@ public function update(Request $request, string $id)
...
             'class_name' => 'required|alpha_num:ascii|max:255',
-            'capacity' => 'required|numeric|integer|min:1', 
+            'capacity' => 'required|numeric|integer|min:1',
             'price' => 'required|decimal:0,2',
             'time_From' => 'required|date|after:tomorrow',
             'time_to' => 'required|date|after:time_From',
+            'image' => 'mimes:png,jpg,jpeg|max:2048',
         ]);
+      
+        $file_name = ($request['image'] == null) ? $request['old_image']:$this->uploadFile($request['image'], 'assets/images');
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ClassController.php:85]: although it is not wrong, but you can benefit from if you didn't send a specific field, it didn't change

what i mean by this 
for example 

if  you send the following data to `update()` method of the model
```php
$data = [
    'class_name' => 'test',
    'capacity' => 10,
    'price' => 13,
];
```

the `time_Form`, `time_to` & `image` won't update

File: app/Traits/Common.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?php
+namespace App\Traits;
+
+trait Common
```
Comment (Nora-Lootfy) [inline app/Traits/Common.php:4]: I am assuming you know what is that in model

```php
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class ModelName extends Model
{
    use HasFactory;
    use SoftDeletes;

    ...
```

File: resources/views/class_details.blade.php [hunk 5]
```diff
@@ -26,7 +26,7 @@
...
+              <img src="{{asset($class['image'])}}"
```
Comment (Nora-Lootfy) [inline resources/views/class_details.blade.php:29]: nice 😊

**Review by Nora-Lootfy — APPROVED** [review]
As usual, amazing

File: .gitignore (hunk 1 above)
Comment (Engy-Mohamed) [reply .gitignore:6]: thank you very much

**Pull request merged**